ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -325,6 +328,7 @@ void poll(RequestRegistrationCallback<GetOperation> requestRegistrationCallback)
       }
       if (firstChunk.isComplete()) {
         onChunkOperationComplete(firstChunk);
+        firstChunk.saveMetadata();
```

Comment:
this is not the best place to call saveMetadata. poll method will be invoked by network client every 50 ms, so we would call saveMetadata over and over again. I recommend calling saveMetadata in onChunkOperationComplete method after we check operationCallbackInvoked.

Reply:
we have a atomicboolean to safe guard the code to be only executed once.